src/Sentry/HubExtensions.cs

```diff
@@ -247,6 +244,11 @@ internal static ITransactionTracer StartTransaction(
 
     internal static ITransactionTracer? GetTransaction(this IHub hub)
     {
+        if (hub is Hub fullHub)
```

Comment:
praise: nice special-case (to reduce closure allocation) and consolidating in `StartSpan`